slo-monitor/README.md

```diff
@@ -0,0 +1,15 @@
+# Kubernetes Performance SLO monitor
+
+This tool monitors Performance SLOs that are unavailable from the API server and exposes Prometheus metrics for them.
+
+It can be run anywhere and only requirement is that it needs to be able to talk with the API server.
+
+# Usage
+
+SLO monitor is a simple pod that needs to be able to read Pods and Events. On top of that it should work as long as it can communicate with the API server.
+
+When starting the monitor you need to set memory limit, as by design it has to be restarted from time to time. It is so because we had two choices:
+- use full-blown informer with the backing store, so we'd be able to handle deletes (and thus keep in useful memory only important Pods) or
+- use a simple Controller that doesn't have backing store, thus can't handle deletes (thus we don't know when it's safe to remove information about a Pod)
```

Comment:
Actually, I looked into the code and it seems that is not true.
So the reflector itself, is calling Add, Update, Delete based on the event type it received (i.e. it doesn't look into contents of the underlying store):
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L372

That means that if you create a simple store with function like:
```
func Add(obj interface) {
  // append to slice ("update", obj)
}
func Update(obj interface)
  // append to slice ("update", obj)
}
...
```
and that will be implementing Queue interface

you will be able to create the Informer by just reimplementing the "NewInformer" function.
[i.e. your clientState should then be nil, the you should redefine the Process function there]

With that, you will be receiving all necessary watch event and you will exactly know whether these are add update or delete. Which is exactly what you want.